docs/internal/GeneralArchitectureGuide.md

```diff
@@ -68,6 +68,43 @@ state must ever be reloaded from persisted state.
 
 ## Deprecations
 
+## Backwards Compatibility
+
+Elasticsearch has major versions (e.g. 8.0.0), minor versions (e.g. 8.3.0) and patch versions (e.g. 8.3.1). Multiversion
+binary backwards compatibility within a cluster is guaranteed for specific situations.
+
+### Transport Layer Backwards Compatibility
+
+Elasticsearch nodes are backwards compatible with all earlier binary version nodes in the same major release: all the earlier
```

Comment:
"are backwards compatible" seems overly general, I recommend spelling out in exactly what way you mean here: can intercommunicate over the network.